graphql/src/schema/api.rs

```diff
@@ -381,11 +364,14 @@ fn field_list_filter_input_values(
                     )
                 }
             }
-            TypeDefinition::Interface(_) => {
+            TypeDefinition::Interface(parent) => {
```

Comment:
apologies if I'm missing something (relatively new to rust) but this expression can be combined w line 357 into `TypeDefinition::Object(parent) | TypeDefinition::Interface(parent) => ` as in line 231 i think? both of these match expressions (L358-366, L368-375) look the same to me

Reply:
`parent` is different in both and I think Rust does not support unions. If I would merge them and try to access `parent.name` I would get an error.

Reply:
I see -- so it works in L231 bc you're discarding the object anyway and here the compiler ~can't understand they both have `.name` independently? and so in this case it would need to use .. some kind of trait?

appreciate the response, thanks!

Reply:
Nice catch - you can make this work if you destructure further:
```rust
            TypeDefinition::Object(ObjectType { name, .. })
            | TypeDefinition::Interface(InterfaceType { name, .. }) => {
                if ast::get_derived_from_directive(field).is_some() {
                    (None, Some(name.clone()))
                } else {
                    (Some(Type::NamedType("String".into())), Some(name.clone()))
                }
            }
```